app/src/org/commcare/android/adapters/EntitySubnodeListAdapter.java

```diff
@@ -0,0 +1,111 @@
+package org.commcare.android.adapters;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ListAdapter;
+
+import org.commcare.android.models.Entity;
+import org.commcare.android.util.AndroidUtil;
+import org.commcare.android.view.EntityView;
+import org.commcare.dalvik.R;
+import org.commcare.suite.model.Detail;
+import org.javarosa.core.model.instance.TreeReference;
+
+import java.util.List;
+
+/**
+ * Created by jschweers on 8/24/2015.
+ * <p/>
+ * Adapter for taking a nodeset, contextualizing it against an entity,
+ * and then displaying one item for each node in the resulting set.
+ */
+public class EntitySubnodeListAdapter implements ListAdapter {
+
+    private Context context;
+    private Detail detail;
+    private List<TreeReference> references;
+    private List<Entity<TreeReference>> entities;
+    private int[] rowColors;
+
+    public EntitySubnodeListAdapter(Context context, Detail detail, List<TreeReference> references, List<Entity<TreeReference>> entities) {
+        this.context = context;
+        this.detail = detail;
+        this.references = references;
+        this.entities = entities;
+
+        this.rowColors = AndroidUtil.getThemeColorIDs(this.context,
+                new int[]{R.attr.drawer_pulldown_even_row_color, R.attr.drawer_pulldown_odd_row_color});
+    }
+
+    @Override
+    public boolean areAllItemsEnabled() {
+        return false;
+    }
+
+    @Override
+    public boolean isEnabled(int position) {
+        return false;
+    }
+
+    @Override
+    public int getCount() {
+        return references.size();
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return references.get(position);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return position;
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return 0;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        EntityView view = (EntityView) convertView;
+        Entity entity = this.entities.get(position);
+        if (view == null) {
+            view = new EntityView(context, detail, entity, null, null, position, false);
+        } else {
+            view.refreshViewsForNewEntity(entity, false, position);
+        }
+        int color = this.rowColors[position % 2];
+        if (color != -1) {
+            view.setBackgroundColor(color);

```

Comment:
The striping in the regular entity detail activity is several times more complicated than this...not sure why. Tested that this works fine as you scroll and views get reused.